repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphRetriever.java

```diff
@@ -1684,7 +1684,15 @@ private AtlasStruct mapVertexToStruct(AtlasVertex entityVertex, String edgeLabel
 
         if (GraphHelper.elementExists(edge)) {
             final AtlasVertex referenceVertex = edge.getInVertex();
-            ret = new AtlasStruct(getTypeName(referenceVertex));
+
+            String typeName = getTypeName(referenceVertex);
+
+            if (StringUtils.isEmpty(typeName)) {
+                LOG.error("typeName not found for vertex {}", getGuid(referenceVertex) );
+                throw new AtlasBaseException(AtlasErrorCode.TYPE_NAME_NOT_FOUND, typeName);
```

Comment:
If typename is not found for a vertex, should we throw an exception or just log it and continue for now? Please check this with Nikhil/Suman once and confirm. If we throw exception here, end user behaviour would be the same, just with a different error message this time I believe.

Reply:
@checkaayush  if typeName is not valid we throw same exception. If we continue the request there will be NPE.
I will evaluate about the error message

Reply:
That's the point. If there is bound to be an exception thrown, then end user behaviour doesn't change. What are we trying to solve with the change then? Idea is if there is some corruption which is non-fatal, should we be throwing an exception, or just log it and move forward? If we have understanding of how we reached this state, only then we can figure out how to prevent it.

Reply:
@aarshi0301  lets discuss this, I agree with @checkaayush  here, first lets find out if vertex is valid if not jut skip it with a log.

cause just changing the error pattern won't help here. After we detect it, We kind of will do the manual skipping anyway to remove the stale index if we do this from code it from itself will help mitigate the error in the first place.